plugin/sampling/strategystore/static/strategy_store.go

```diff
@@ -216,6 +233,49 @@ func (h *strategyStore) parseStrategies(strategies *strategies) {
 		newStore.defaultStrategy = h.parseServiceStrategies(strategies.DefaultStrategy)
 	}
 
+	for _, s := range strategies.ServiceStrategies {
+		newStore.serviceStrategies[s.Service] = h.parseServiceStrategies(s)
+
+		// Config for this service may not have per-operation strategies,
+		// but if the default strategy has them they should still apply.
+
+		if newStore.defaultStrategy.OperationSampling == nil {
+			// Default strategy doens't have them either, nothing to do.
+			continue
+		}
+
+		opS := newStore.serviceStrategies[s.Service].OperationSampling
+		if opS == nil {
+
+			// Service does not have its own per-operation rules, so copy (by value) from the default strategy.
+			newOpS := *newStore.defaultStrategy.OperationSampling
+
+			// If the service's own default is probabilistic, then its sampling rate should take precedence.
+			if newStore.serviceStrategies[s.Service].ProbabilisticSampling != nil {
+				newOpS.DefaultSamplingProbability = newStore.serviceStrategies[s.Service].ProbabilisticSampling.SamplingRate
+			}
+			newStore.serviceStrategies[s.Service].OperationSampling = &newOpS
+			continue
+		}
+
+		// If the service did have its own per-operation strategies, then merge them with the default ones.
+		opS.PerOperationStrategies = mergePerOperationSamplingStrategies(
+			opS.PerOperationStrategies,
+			newStore.defaultStrategy.OperationSampling.PerOperationStrategies)
+	}
+	h.storedStrategies.Store(newStore)
+}
+
+func (h *strategyStore) parseStrategies_deprecated(strategies *strategies) {
```

Comment:
to minimize code changes diff, I would prefer if you renamed the original function to parseStrategies_deprecated (leaving it in the same place where it's defined now) and added the new function below. This would make the diff more readable and easier to see that the original function did not change. Not it all shows as changed / added, nothing to compare against.